### PR TITLE
Multiple routes

### PR DESCRIPTION
**Pull request by rickaha**

### CODE REVIEW

File: autopilot/multiwaypointfollower.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+{
+    mWayPointFollowerList[mRouteID]->resetState();
+}
+
+void MultiWaypointFollower::addWaypointFollower(QSharedPointer<WaypointFollower> waypointfollower)
+{
+    mWayPointFollowerList.append(waypointfollower);
+}
+
+void MultiWaypointFollower::setActiveRoute(int routeID)
```
Comment (marvind) [inline autopilot/multiwaypointfollower.cpp:68]: Should be called 'setActivewaypointFollower(int waypointfollowerID' IMO to match the 'addWaypointFollower' and 'getActiveWaypointFollower' function. It is actually a waypointfollower that is set active, not a route.

File: autopilot/multiwaypointfollower.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+{
+    mRouteID = routeID;
+}
+
+QSharedPointer<WaypointFollower> MultiWaypointFollower::getActiveWaypointFollower()
+{
+    return mWayPointFollowerList[mRouteID];
+}
+
+int MultiWaypointFollower::getNumberOfRoutes()
```
Comment (marvind) [inline autopilot/multiwaypointfollower.cpp:78]: getNumberOfWaypointfollowers()

File: autopilot/multiwaypointfollower.cpp [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+void MultiWaypointFollower::resetState()
+{
+    mWayPointFollowerList[mRouteID]->resetState();
+}
+
+void MultiWaypointFollower::addWaypointFollower(QSharedPointer<WaypointFollower> waypointfollower)
+{
+    mWayPointFollowerList.append(waypointfollower);
```
Comment (marvind) [inline autopilot/multiwaypointfollower.cpp:65]: As far as I understand, the Waypointfollower's ID is determined here. Therefore, I would return it 
-> int MultiWaypointFollower::addWaypointFollower(..)

**Review by marvind — APPROVED** [review]
Looks good! Minor changes suggested.

File: autopilot/multiwaypointfollower.cpp (hunk 1 above)
Comment (rickaha) [reply autopilot/multiwaypointfollower.cpp:68]: I agree

**Pull request merged**